rules/xcodeproj.bzl

```diff
@@ -785,6 +858,38 @@ Tags for configuration:
     xcodeproj-ignore-as-target: Add this to a rule declaration so that this rule will not generates a scheme for this target
 """,
     attrs = {
+        "build_configurations_debug": attr.string_list(
+            mandatory = False,
+            default = [],
+            doc = "List of names for build configurations of type 'debug' to generate. If 'Debug' is not present it will be added.",
+        ),
+        "build_configurations_debug_bazelrc": attr.string_dict(
+            mandatory = False,
+            default = {},
+            doc = """
+        Dictionary where each key is the name of a build configuration in 'build_configurations_debug'
+        and the 'value' is the name of the respective configuration in the .bazelrc file.
+
+        Set this attribute if you want the generated Xcode project to append '--config value' when
+        building for each specified build configuration
+        """,
+        ),
+        "build_configurations_release": attr.string_list(
```

Comment:
What do you think about using a dict and taking it's keys as the configs instead of putting the keys in both places?

Also, if possible I'd try to keep away from `Debug` and `Release` idioms to reduce the possible build permutations and allow any config. I'm not sure what this implicates inside of XcodeGen but maybe internally the rule can just use debug and remove any defaults - I'm not sure we'd want XcodeGen defaults anyhow.

Reply:
I think xcodegen accepts other values different than `debug`/`release` and it simply doesn't apply any value by default (learned that today tbh).

I'll see if I can remove the `debug`/`release` idiom here, just call it `build_configurations` and see how to separate things inside